contracts/gateway/EIP20Gateway.sol

```diff
@@ -964,7 +980,56 @@ contract EIP20Gateway is Gateway {
         message.gasConsumed = initialGas.sub(gasleft());
     }
 
-    /* Private functions */
+    /**
+    * @notice Activate Gateway contract. Can be set only by the
+    *         Organisation address only once by passing co-gateway address.
+    *
+    * @param _coGatewayAddress Address of cogateway.
+    *
+    * @return `true` if value is set
+    */
+    function activateGateway(address _coGatewayAddress)
+        external
+        onlyOrganisation
+        returns (bool)
+    {
+
+        require(
+            remoteGateway == address(0),
+            'Gateway was already activated once.'
+        );
+
+        remoteGateway = _coGatewayAddress;
```

Comment:
Why not `coGateway` in storage or `_remoteGateway` as argument?

Reply:
remoteGateway storage is defined in `GatewayBase`, which can be both gateway and co-gateway. 
But here `activateGateway` method is defined in `EIP20Gateway` context which means remote gateway is know here which is `Co-gateway`.